src/section.rs

```diff
@@ -69,13 +98,13 @@ pub trait WritePiggybackSection {
     }
 }
 
-fn read_entry(entry: Result<DirEntry, Error>) -> AnyhowResult<Section> {
+fn read_entry(entry: Result<DirEntry, walkdir::Error>) -> AnyhowResult<Section> {
     let entry = entry?;
     let raw = read_to_string(entry.path())?;
     Ok(serde_json::from_str(&raw)?)
 }
 
-pub fn read(directory: impl AsRef<Path>, locker: &Locker) -> AnyhowResult<Vec<Section>> {
+pub fn read(directory: impl AsRef<Path>, locker: &Locker) -> Result<Vec<Section>, WriteError> {
```

Comment:
Wouldn't `Result<Vec<Section>, LockerError>` be sufficient here?

Reply:
The question mark operator only calls the `From` trait once. However, my implementation requires two `from` calls to turn Locker Errors into SetupErrors, and only one to convert a WriteError. Thus, in the setup functions working with the WriteError is easier.

Reply:
It seems strange that a function called `read` returns a `WriteError`. If possible, please us `LockerError` instead.

Reply:
Done